src/Ledger/Conway/Specification/Epoch.lagda

```diff
@@ -190,84 +192,110 @@ described in \textcite[\sectionname~6.4]{shelley-delegation-design}.
+opaque
+  createRUpd : ℕ → BlocksMade → EpochState → Coin → RewardUpdate
+  createRUpd slotsPerEpoch b es total =
+    record  { Δt = Δt₁
+            ; Δr = 0 - Δr₁ + Δr₂
+            ; Δf = 0 - pos feeSS
+            ; rs = rs
 \end{code}
 \begin{code}[hide]
-          ; flowConservation = flowConservation
-          ; Δt-nonnegative = Δt-nonneg
-          ; Δf-nonpositive = Δf-nonpos
+            ; flowConservation = flowConservation
+            ; Δt-nonnegative = Δt-nonneg
+            ; Δf-nonpositive = Δf-nonpos
 \end{code}
 \begin{code}
-          }
-  where
-    prevPp       = PParamsOf es
-    reserves     = reservesOf es
-    pstakego     = es .EpochState.ss .Snapshots.go
-    feeSS        = es .EpochState.ss .Snapshots.feeSS
-    stake        = pstakego .Snapshot.stake
-    delegs       = pstakego .Snapshot.delegations
-    poolParams   = pstakego .Snapshot.poolParameters
-    blocksMade   = ∑[ m ← b ] m
-    ρ            = fromUnitInterval (prevPp .PParams.monetaryExpansion)
-    η            = fromℕ blocksMade ÷₀ (fromℕ slotsPerEpoch * ActiveSlotCoeff)
-    Δr₁          = floor (1 ⊓ η * ρ * fromℕ reserves)
-    rewardPot    = pos feeSS + Δr₁
-    τ            = fromUnitInterval (prevPp .PParams.treasuryCut)
-    Δt₁          = floor (fromℤ rewardPot * τ)
-    R            = rewardPot - Δt₁
-    circulation  = total - reserves
-    rs           = reward prevPp b (posPart R) poolParams stake delegs circulation
-    Δr₂          = R - pos (∑[ c ← rs ] c)
+            }
+    where
+      prevPp       : PParams
+      prevPp       = PParamsOf es
+      reserves     : Coin
+      reserves     = reservesOf es
+      pstakego     : Snapshot
+      pstakego     = es .EpochState.ss .Snapshots.go
+      feeSS        : Coin
+      feeSS        = es .EpochState.ss .Snapshots.feeSS
+      stake        : Credential ⇀ Coin
+      stake        = pstakego .Snapshot.stake
+      delegs       : Credential ⇀ KeyHash
+      delegs       = pstakego .Snapshot.delegations
+      poolParams   : KeyHash ⇀ StakePoolParams
+      poolParams   = pstakego .Snapshot.poolParameters
+      blocksMade   : ℕ
+      blocksMade   = ∑[ m ← b ] m
+      ρ            : ℚ
+      ρ            = fromUnitInterval (prevPp .PParams.monetaryExpansion)
+      η            : ℚ
+      η            = fromℕ blocksMade ÷₀ (fromℕ slotsPerEpoch * ActiveSlotCoeff)
+      Δr₁          : ℤ
+      Δr₁          = floor (1 ⊓ η * ρ * fromℕ reserves)
+      rewardPot    : ℤ
+      rewardPot    = pos feeSS + Δr₁
+      τ            : ℚ
+      τ            = fromUnitInterval (prevPp .PParams.treasuryCut)
+      Δt₁          : ℤ
+      Δt₁          = floor (fromℤ rewardPot * τ)
+      R            : ℤ
+      R            = rewardPot - Δt₁
+      circulation  : Coin
+      circulation  = total - reserves
+      rs           : Credential ⇀ Coin
+      rs           = reward prevPp b (posPart R) poolParams stake delegs circulation
+      Δr₂          : ℤ
+      Δr₂          = R - pos (∑[ c ← rs ] c)
```

Comment:
I like that you added all the types... but I think we could clean this up using type classes and grouping like types on a single line (e.g., `ρ η τ : ℚ`).

Reply:
I think this has been adopted as proposed in https://github.com/IntersectMBO/formal-ledger-specifications/pull/869#discussion_r2331912931